frontend/simple/model/contracts/group.js

```diff
@@ -0,0 +1,209 @@
+      // ['memberApprovalThreshold', 'float'],
+      // ['memberRemovalThreshold', 'float'],
+      // ['incomeProvided', 'float'],
+      // ['incomeCurrency', 'string'],
+      // ['contributionPrivacy', 'string'],
+      // ['founderUsername', 'string'],
+      // ['founderIdentityContractId', 'string']
+    },
+    vuex: {
+      initialState: { // may be specified in the constructor only
+        payments: [],
+        invitees: [],
+        profiles: {}, // usernames => {contractId: string, groupProfile: Object}
+        proposals: {} // hashes => {} TODO: this, see related TODOs in GroupProposal
+      },
+      mutation: (state, {data}) => {
+        Object.assign(state, {
+          profiles: {
+            [data.founderUsername]: {
+              contractID: data.founderIdentityContractId,
+              groupProfile: {}
+            }
+          }
+        }, data)
+      },
+      getters: {
+        candidateMembers (state) {
+          return Object.keys(state.proposals).filter(key => state.proposals[key].candidate).map(key => state.proposals[key].candidate)
+        },
+        memberUsernames (state) {
+          return Object.keys(state.profiles)
+        }
+      }
+    }
+  },
+  'GroupPayment': {
+    validate: function (data) {
+      // ['payment', 'string'] // TODO: change to 'double' and add other fields
+    },
+    vuex: {
+      mutation: (state, {data}) => { state.payments.push(data) }
+    }
+  },
+  'GroupProposal': {
+    constants: {
+      TypeInvitation: 'invitationProposal',
+      TypeRemoval: 'removalProposal',
+      TypeChange: 'changeProposal'
+    },
+    validate: function (data) {
+      // ['type', 'string'],
+      // ['threshold', 'float'],
+      // ['actions', 'Action', 'repeated'],
+      //   // array of objects of type:
+      //   ['contractID', 'string'],
+      //   ['action', 'string']
+      // ['candidate', 'string'],
+      // ['initiator', 'string'],
+      // ['initiationDate', 'string'],
+      // ['expirationDate', 'string']
+    },
+    vuex: {
+      mutation: (state, {data, hash}) => {
+        // TODO: this should be data instead of ...data to avoid conflict with neighboring properties
+        // TODO: convert to votes instead of for/against for future-proofing
+        state.proposals[hash] = {...data, for: [data.initiator], against: []}
+      }
+    }
+  },
+  // TODO: rename this to just GroupProposalVote, and switch off of the type of vote
+  'GroupVoteForProposal': {
+    validate: function (data) {
+      // ['username', 'string'],
+      // ['proposalHash', 'string']
+    },
+    vuex: {
+      mutation: (state, {data}) => {
+        if (state.proposals[data.proposalHash]) {
+          state.proposals[data.proposalHash].for.push(data.username)
+          let threshold = Math.ceil(state.proposals[data.proposalHash].threshold * Object.keys(state.profiles).length)
+          if (state.proposals[data.proposalHash].for.length >= threshold) {
```

Comment:
`state.proposals[data.proposalHash]` is used 4 times here. what about having a variable?

Reply:
I'm avoiding messing with the old proposals code as I hope @hubudibu will be able to rewrite it and improve upon it, so my messing with it right now probably wouldn't be worthwhile